OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java

```diff
@@ -37,6 +37,11 @@ public class GpxDbHelper implements GpxDbReaderCallback {
 	private final Map<File, GpxDataItemCallback> readingItemsCallbacks = new ConcurrentHashMap<>();
 
 	private GpxReaderTask readerTask;
+	private DBTestCallback dbTestCallback;
```

Comment:
static counter will be enough for test, how many reads are done from DB

Reply:
using callback we ensure we perform this action only in test, checking only specific track and skip analysis update on initial track loading.